### PR TITLE
One-Time Password (OTP) User Verification

### PR DESCRIPTION
**Pull request by PlainOlSoapBar**
## User Model Updated
- verified: Boolean (default: False)
## New UserOTP Model
- userId: String
- otp: String
- createdAt: Date
- expiredAt: Date
## OTP Code
- 6 digits.
- Expires after one hour.
- Sent to the user's email.
- Bcrypted/hashed.
## Routes
- /verify (POST)
  - Inputting the correct OTP will change the user's verified property to True.
  - OTP will be deleted when used.
- /resendVerification (POST)
  - Creates and emails a new OTP, while deleting old ones.
## User Services Updated
- sendUserOTP
  - Function to send and email OTP.
  - Requires AUTH_EMAIL and AUTH_PASS from .env (see below).
## .env File
Requires these two fields to email OTP:
- AUTH_EMAIL = [Bronco Bond Gmail]
- AUTH_PASS = [Bronco Bond Gmail's App Password]

### CODE REVIEW

**Comment by PlainOlSoapBar** [comment]
Several commits from #56 are technically part of this PR.

File: backends/controller/user.controller.js
Comment (Jay7vn) [inline backends/controller/user.controller.js:1]: When resending verification, check if user is already verified

Comment (Jay7vn) [reply backends/controller/user.controller.js:1]: Fix Bug, which allowed for any OTP to be given and allow successful verification

Comment (Jay7vn) [inline backends/controller/user.controller.js:1]: Fix Bug, which allowed for any OTP to be given and allow successful verification

**Review by Jay7vn — APPROVED** [review]

**Review by Jay7vn — APPROVED** [review]

**Pull request merged**